ndc-lambda-sdk/src/execution.ts

```diff
@@ -211,20 +202,76 @@ function buildCausalStackTrace(error: Error): string {
   return stackTrace;
 }
 
-export function reshapeResultToNdcResponseValue(value: unknown, type: schema.TypeDefinition, valuePath: string[], fields: Record<string, sdk.Field> | "AllColumns", objectTypes: schema.ObjectTypeDefinitions): unknown {
+// Represents either selecting a scalar (ie. the whole value, opaquely), an object (selecting properties), or an array (select whole array)
+export type FieldSelection = sdk.NestedField | { type: "scalar" }
```

Comment:
This scalar stanza seems to be used in a few places. Worth having a constant?

Reply:
Oh, I think I understand what you mean (you made a [similar comment](https://github.com/hasura/ndc-nodejs-lambda/pull/10#discussion_r1490711043) on the relaxed types PR). Personally I feel that `{ type: "scalar" }` is small and obvious enough to not need a common const extraction for naming purposes. 

As for memory sharing purposes, I get a bit uncomfortable with sharing objects as "constants" because they're still mutable, and bad code could mutate the "constant" and corrupt every place it's reused. That's me being a bit anal, I admit.

However, as I said on the other comment, if you feel very strongly about this, I'll change it.

Reply:
I don't mind if you prefer it this way.